walkthrough/step04.md

```diff
@@ -0,0 +1,217 @@
+# Step04 - Encoding cookies with JWT
+
+Now we have a user logging in and then setting the cookie, and from there we recognize the user through that cookie, but anyone can change that cookie. which is a big security vulnerability.
+
+
+`Token-based authentication` is a way to authenticate a user. The idea is that upon login we create a token (a string) and store it in a cookie. The browser then sends the token with subsequent requests (since cookies are sent with every request), and if the token can be verified (i.e. it is a legitimate token issued by the server), the user is authenticated.
+
+**would love to have an image here but couldn't find any good ones for tokens :smile: issues are welcomed**
```

Comment:
How about dis? https://d33wubrfki0l68.cloudfront.net/746c5467430de460750e7fc19963f462d120e018/b1f80/assets-jekyll/blog/what-happens-if-your-jwt-is-stolen/stolen-jwt-e0fbf1b4e58f692decd4f046ac66f1daca94a369bfa6375e7837509e15317c43.png